File: Graph/Python/11724 연결 요소의 개수.py
from collections import deque

graph = [[] for i in range(1001)]
visited = [False for i in range(1001)]
n, m = map(int, input().split())
answer = 0
for i in range(m) :
    f,t = map(int, input().split())
    graph[f].append(t)
    graph[t].append(f)
    
def bfs(start) :
    if visited[start] : return False
    queue = deque()
    visited[start] = True
    queue.append(start)
    while queue :
        v = queue.popleft()
        for to in graph[v] :
            if not visited[to] :
                visited[to] = True
                queue.append(to)
    return True


for i in range(1, n+1) :
    if bfs(i) :
        answer += 1
        
print(answer)

File: mvc1/_etc/Java/11724 연결 요소의 개수.java
import java.util.Scanner;
import java.util.*;

public class Main {
    
    static ArrayList<Integer>[] graph = new ArrayList[1001];
    static boolean[] visited = new boolean[1001];
    
    static boolean bfs(int start) {
        if(visited[start] == true) return false;
        Queue<Integer> queue = new LinkedList();
        visited[start] = true;
        queue.add(start);
        
        while(!queue.isEmpty()) {
            int v = queue.peek(); queue.poll();
            for(int to : graph[v]) {
                if(visited[to] == false) {
                    visited[to] = true;
                    queue.add(to);
                }
            }
        }
        
        return true;
        
    }
    
    
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        for(int i = 0 ; i < 1001; i++) graph[i] = new ArrayList ();
        int n = sc.nextInt(), m = sc.nextInt();
        for(int i = 0 ; i < m; i++) {
            int from = sc.nextInt(), to = sc.nextInt();
            graph[from].add(to); graph[to].add(from);
        }
        int answer = 0;
        for(int i = 1; i < n+1; i++) {
            if(bfs(i)) answer += 1;
        }
        
        System.out.print(answer);
        
    }
}
